src/clients/FtpGateway.cc

```diff
@@ -704,18 +702,22 @@ ftpListParseParts(const char *buf, struct Ftp::GatewayFlags flags)
                 p->size = atoi(ct + 1);
                 break;
 
-            case 'm':
-                tm = (time_t) strtol(ct + 1, &tmp, 0);
-
-                if (tmp != ct + 1)
+            case 'm': {
+                char *tmp;
+                const auto tm = (time_t) strtol(ct + 1, &tmp, 0);
+                if (tmp == ct + 1)
                     break;  /* not a valid integer */
```

Comment:
If my suggestion in another change request is accepted, then this can become an assertion:

```suggestion
                Assure(tmp > ct + 1); // saw at least one digit
```

To polish further, `ct + 1` should be given a name (e.g., secondsStart) and `tmp` should be renamed to something like secondsEnd. There are also missed AAA and `const` opportunities in modified code.